### ISSUE TITLE
Strange LESS file message

### ISSUE THREAD

**Issue by smlombardi** [AtomLinter/linter-stylelint#181]
I was editing my theme and got this:

![screen shot 2016-04-27 at 8 08 09 am](https://cloud.githubusercontent.com/assets/835148/14852053/0217da20-0c52-11e6-8055-13b1fa826427.jpg)

**Comment by Arcanemagus** [comment]
Can you upload the raw file? Does `stylelint` itself show that error if you run from the CLI?

**Comment by smlombardi** [comment]
i can't get it to happen again.  I will keep an eye out.

**Comment by Arcanemagus** [comment]
Okay, marking as Unable to reproduce and closing for now.

**Comment by gregorywyz** [comment]
This happens when I create a new less file and attempt to create a new classname. The error is triggered just before typing the opening bracket

**Comment by Arcanemagus** [comment]
Managed to reproduce this _intermittently_ from your steps @gregorywyz. Got the following stack trace which tells us that this is a bug in `postcss-less`:

```
TypeError: Cannot read property '2' of undefined
    at LessParser.rule (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss-less\dist\less-parser.js:64:38)
    at LessParser.createRule (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss-less\dist\less-parser.js:110:18)
    at LessParser.ruleWithoutBody (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss-less\dist\less-parser.js:141:18)
    at LessParser.processEndOfRule (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss-less\dist\less-parser.js:173:22)
    at LessParser.word (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss-less\dist\less-parser.js:272:44)
    at LessParser.loop (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss-less\dist\less-parser.js:300:30)
    at lessParse (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss-less\dist\less-parse.js:23:12)
    at new LazyResult (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss\lib\lazy-result.js:70:24)
    at Processor.process (C:\Users\Landon Abney\.atom\packages\linter-stylelint\node_modules\postcss\lib\processor.js:117:12)
    at file:///C:/Users/Landon Abney/Documents/GitHub-release/linter-stylelint/node_modules/stylelint/lib/getPostcssResult.js:81:29
```

**Comment by Arcanemagus** [comment]
Filed an issue there, hopefully it can be taken care of when somebody takes on that project again.

**Comment by shellscape** [comment]
@Arcanemagus @gregorywyz I've been given ownership of postcss-less and will be actively maintaining it for the foreseeable future. If you all can get me a file or line that produces an error consistently using the parsing methods [shown here](https://github.com/shellscape/postcss-less/#rulerulewithoutbody), I can probably debug it. Unfortunately I don't have a whole bunch of interest in debugging it through Atom, through the plugin system, and through the plugin itself 😄

**Comment by Arcanemagus** [comment]
@shellscape I'll see what I can do, it's an intermittent issue though so reproducing it out of that environment will likely be difficult.

**Comment by Arcanemagus** [comment]
Also, I would much rather keep discussion over in https://github.com/shellscape/postcss-less/issues/64 as the bug has nothing to do with this package's code.

**Comment by abea** [comment]
I see in https://github.com/shellscape/postcss-less/issues/64 that this should be resolved on that end as well as in this linter. My linter-stylelint package is up to date but I'm still getting this error when writing a new selector. Is there something I'm missing or that isn't yet updated?

**Comment by shellscape** [comment]
@abea stylelint may not have updated to the version which has that fix, or the linter plugin may not be following the pattern of trapping that error as a user types. @Arcanemagus that's probably something this plugin will have to do, rather than the stylelint package - you may wish to reopen this one.

**Comment by Arcanemagus** [comment]
Looks like `stylelint` only updated `postcss-less` in https://github.com/stylelint/stylelint/pull/2702, which will be included in `stylelint@8`.
